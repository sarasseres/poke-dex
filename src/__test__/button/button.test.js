import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../../components/Button';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button></Button>, div);
});

test('renders button correctly', () => {
  const { getByTestId } = render(<Button label='Simpan'></Button>);
  expect(getByTestId('button')).toHaveTextContent('Simpan');
});

test('renders button correctly', () => {
  const { getByTestId } = render(<Button label='Batal'></Button>);
  expect(getByTestId('button')).toHaveTextContent('Batal');
});

test('matches snapshot 1', () => {
  const tree = renderer.create(<Button label='Simpan'></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches snapshot 2', () => {
  const tree = renderer.create(<Button label='Batal'></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});