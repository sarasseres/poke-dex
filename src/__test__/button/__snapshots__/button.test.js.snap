// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained css-1o1v0on-GeneralButton MuiButton-containedSecondary MuiButton-disableElevation"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Simpan
  </span>
</button>
`;

exports[`matches snapshot 1 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained css-1o1v0on-GeneralButton MuiButton-containedSecondary MuiButton-disableElevation"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Simpan
  </span>
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained css-1o1v0on-GeneralButton MuiButton-containedSecondary MuiButton-disableElevation"
  data-testid="button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Batal
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
